eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -401,34 +398,93 @@ public void testInvalidConfigJSON() {
     }
 
     @Test
-    public void testCachedBadResponseAllowsLaterFetching() {
+    public void testCachedBadResponseRequiresFetch() {
         // Populate the cache with a bad response
-        ConfigCacheFile cacheFile = new ConfigCacheFile(ApplicationProvider.getApplicationContext());
-        cacheFile.delete();
-        try {
-            cacheFile.getOutputWriter().write("{}");
-            cacheFile.getOutputWriter().close();
-        } catch (IOException e) {
-            throw new RuntimeException(e);
-        }
+        ConfigCacheFile cacheFile = new ConfigCacheFile(ApplicationProvider.getApplicationContext(), safeCacheKey(DUMMY_API_KEY));
+        cacheFile.setContents("{ invalid }");
 
-        initClient(TEST_HOST, false, false, false);
+        initClient(TEST_HOST, true, false, false, DUMMY_API_KEY);
 
-        String result = EppoClient.getInstance().getStringAssignment("dummy subject", "dummy flag");
-        assertNull(result);
-        // Failure callback will have fired from cache read error, but configuration request will still be fired off on init
-        // Wait for the configuration request to load the configuration
-        waitForNonNullAssignment();
-        String assignment = EppoClient.getInstance().getStringAssignment("6255e1a7fc33a9c050ce9508", "randomization_algo");
-        assertEquals("control", assignment);
+        String assignment = EppoClient.getInstance().getStringAssignment("6255e1a7d1a3025a26078b95", "randomization_algo");
```

Comment:
no need to wait as the completed callback should not be called until fetch succeeds